code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplateTest.kt

```diff
@@ -0,0 +1,124 @@
+            BasicPushTemplate(MapData(aepPushBasicData))
+        }
+        assertEquals(
+            "Required push template key ${PushTemplateConstants.PushPayloadKeys.TITLE} not found or null",
+            exception.message
+        )
+    }
+
+    @Test
+    fun `Test AEPPushTemplate initialization with missing body`() {
+        val aepPushData = MockAEPPushTemplateDataProvider.getMockedDataMapWithRequiredData()
+        aepPushData.remove(PushTemplateConstants.PushPayloadKeys.BODY)
+        val exception = assertFailsWith<IllegalArgumentException> {
+            BasicPushTemplate(MapData(aepPushData))
+        }
+        assertEquals(
+            "Required push template key ${PushTemplateConstants.PushPayloadKeys.BODY} not found or null",
+            exception.message
+        )
+    }
+
+    @Test
+    fun `Test AEPPushTemplate initialization with missing version`() {
+        val aepPushData = MockAEPPushTemplateDataProvider.getMockedDataMapWithRequiredData()
+        aepPushData.remove(PushTemplateConstants.PushPayloadKeys.VERSION)
+        val exception = assertFailsWith<IllegalArgumentException> {
+            BasicPushTemplate(MapData(aepPushData))
+        }
+        assertEquals(
+            "Required push template key ${PushTemplateConstants.PushPayloadKeys.VERSION} not found or null",
+            exception.message
+        )
+    }
+
+    @Test
+    fun `Test AEPPushTemplate initialization with all data`() {
+        val aepPushTemplate = provideMockedBasicPushTemplateWithAllKeys()
+        assertEquals(PushTemplateType.BASIC, aepPushTemplate.templateType)
+        assertEquals(MOCKED_TAG, aepPushTemplate.tag)
+        assertFalse(aepPushTemplate.isFromIntent)
+        assertEquals(MOCKED_ACTION_URI, aepPushTemplate.actionUri)
+        assertEquals(PushTemplateConstants.ActionType.NONE, aepPushTemplate.actionType)
+        assertEquals(2, aepPushTemplate.actionButtonsList?.size)
+        assertEquals(5, aepPushTemplate.badgeCount)
+        assertEquals(MOCKED_BASIC_TEMPLATE_BODY, aepPushTemplate.body)
+        assertEquals("2024", aepPushTemplate.channelId)
+        assertEquals(MOCKED_BASIC_TEMPLATE_BODY_EXPANDED, aepPushTemplate.expandedBodyText)
+        assertEquals("FFD966", aepPushTemplate.bodyTextColor)
+        assertEquals(MOCKED_IMAGE_URI, aepPushTemplate.imageUrl)
+        assertEquals(MOCKED_LARGE_ICON, aepPushTemplate.largeIcon)
+        assertEquals("FFD966", aepPushTemplate.backgroundColor)
+        assertEquals(MOCKED_PRIORITY, aepPushTemplate.getNotificationPriority())
+        assertEquals(
+            NotificationVisibility.getNotificationCompatVisibilityFromString(
+                MOCKED_VISIBILITY
+            ),
+            aepPushTemplate.getNotificationVisibility()
+        )
+        assertEquals("PRIORITY_DEFAULT", aepPushTemplate.priorityString)
+        assertEquals("PUBLIC", aepPushTemplate.visibilityString)
+        assertEquals("remind me", aepPushTemplate.remindLaterText)
```

Comment:
Remind later is part of BasicPushTemplate, please move this to `BasicPushTemplateTests.kt`

Reply:
Done removed, these assertions exist in BasicPushTemplate.